tests/unit/test_query_language.py

```diff
@@ -381,3 +384,36 @@ def test_incompatible_duration_operations(lhs, op, rhs):
         else:
             assert False
         assert not result
+
+
+@pytest.mark.parametrize(
+    "fn_name",
+    {
+        k
+        for k, v in DateFunctions.__dict__.items()
+        if callable(v) and "add_" not in k and "sub_" not in k
+    }
+    | {
+        k
+        for k, v in BaseSeries.__dict__.items()
+        if callable(v) and "__" not in k and "map" not in k
+    },
+)
+def test_ehrql_date_string_equivalence(fn_name):
+    date_str = "2000-01-01"
+    date_date = date(2000, 1, 1)
```

Comment:
Micro nit: since these are only used once each, why not inline them?  It's one less thing for the reader to keep in L0 cache.